spring/src/main/java/io/seata/spring/annotation/GlobalTransactionScanner.java

```diff
@@ -287,6 +283,17 @@ public void afterPropertiesSet() {
         }
         initClient();
 
+        setCustomFailureHandler();
+    }
+
+    private void setCustomFailureHandler() {
+        if (failureHandlerHook == null) {
+            try {
+                failureHandlerHook = applicationContext.getBean(FailureHandler.class);
```

Comment:
test failed, FailureHandler has not yet been instantiated.

Reply:
Unit tests have passed,Is the custom FailureHandler in the container?If there is no customization, FailureHandler  is null,Because GlobalTransactionalInterceptor has a default

Reply:
Custom FailureHandler implementation need be instantiated before GlobalTransactionScanner initialization.
Unit tests initialize the GlobalTransactionScanner first.